### PR TITLE
update RELEASES.md to include release checklist

### PR DESCRIPTION
**Pull request by a-mccarthy**
Signed-off-by: Abigail McCarthy <mabigail@vmware.com>

This PR updates the RELEASE.md file to include some more detailed steps around the non-code pieces of a release.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -26,3 +26,23 @@ The Harbor project maintains release branches for the three most recent minor re
...
+1. Prepare information about what's new in the release.
+
+  * Update documentation for changes that have happened in the release. See the [goharbor/website](https://github.com/goharbor/website) repo for more details on how to create documentation for a release. All documentation for a release should be published by the time the release is out.
+  * Write release notes. See [previous releases](https://github.com/goharbor/harbor/releases) for examples of what to included in release notes.
+  * For a major/minor release, write a blog post that highlights new features in the release. Plan to publish this the same day as the release. Highlight the themes, or areas of focus, for the release. Some examples of themes are security, bug fixes, feature improvements. If there are any new features or workflows introduced in a release, consider writing additional blog posts to help users learn about the new features. Plan to publish these after the release date (all blogs don’t have to be publish all at once).
+
+
+3. Release a new version. Make the new version, docs updates, and blog posts available.
+
+4. Announce the release and thank contributors.
```
Comment (jonasrosland) [inline RELEASES.md:43]: The item numbers seem to be incorrect, if you remove the whitespaces you can use `1.` everywhere to alleviate any numbering concerns.

File: RELEASES.md [hunk 2]
```diff
@@ -26,3 +26,23 @@ The Harbor project maintains release branches for the three most recent minor re
 
 ### Next Release
 The activity for next release will be tracked in the [up-to-date project board](https://github.com/orgs/goharbor/projects/1). If your issue is not present in the corresponding release, please reach out to the maintainers to add the issue to the project board.
+
+### Publishing a New Release
+
+The following steps outline what to do when its time to plan for and publish a release. Depending on the release (major/minor/patch), not all the following items are needed.
```
Comment (jonasrosland) [inline RELEASES.md:32]: Can we be more specific about what is excluded from say a patch that needs to be included in a minor?

File: RELEASES.md [hunk 3]
```diff
@@ -26,3 +26,23 @@ The Harbor project maintains release branches for the three most recent minor re
...
+  * Write release notes. See [previous releases](https://github.com/goharbor/harbor/releases) for examples of what to included in release notes.
+  * For a major/minor release, write a blog post that highlights new features in the release. Plan to publish this the same day as the release. Highlight the themes, or areas of focus, for the release. Some examples of themes are security, bug fixes, feature improvements. If there are any new features or workflows introduced in a release, consider writing additional blog posts to help users learn about the new features. Plan to publish these after the release date (all blogs don’t have to be publish all at once).
+
+
+3. Release a new version. Make the new version, docs updates, and blog posts available.
+
+4. Announce the release and thank contributors.
+
+  * In all messages to the community include a brief list of highlights and links to the new release blog, release notes, or download location. Also include shoutouts to community memeber contribution included in the release.
```
Comment (jonasrosland) [inline RELEASES.md:45]: memeber > member

File: RELEASES.md [hunk 4]
```diff
@@ -26,3 +26,23 @@ The Harbor project maintains release branches for the three most recent minor re
...
+  * Send an email to the community via the mailing list
```
Comment (jonasrosland) [inline RELEASES.md:46]: Can you add a link to the mailing list?

File: RELEASES.md [hunk 5]
```diff
@@ -26,3 +26,23 @@ The Harbor project maintains release branches for the three most recent minor re
...
+  * Post a message in the Harbor [slack channel](https://app.slack.com/client/T08PSQ7BQ/CC1E09J6S)
```
Comment (jonasrosland) [inline RELEASES.md:47]: I have the channel link as https://cloud-native.slack.com/archives/CC1E09J6S, which one is correct?

**Review by jonasrosland — CHANGES_REQUESTED** [review]
A few small changes, otherwise LGTM!

File: RELEASES.md (hunk 5 above)
Comment (a-mccarthy) [reply RELEASES.md]: probably yours :)  i updated it to bee the one you have

**Review by jonasrosland — APPROVED** [review]
LGTM!

**Review by xaleeks — APPROVED** [review]
LGTM

**Pull request merged**